internal/pkg/gateway/registry.go

```diff
@@ -127,19 +127,29 @@ func (reg *registry) endorsers(channel string, chaincode string) ([]*endorser, e
 		for _, peer := range receivers {
 			endorsers = append(endorsers, peer.endorser)
 		}
+
+		// if this plan doesn't contain the local org, abandon it in favour of one that does, since we already have a local endorsement
```

Comment:
```suggestion
		// if this plan doesn't contain the `preferOrg` org, abandon it in favour of one that does, since we already have a local endorsement
```
The term `local org` confused me for a while until I saw the usage of this function. 
As far as this function is concerned, preferOrg is supplied and in some case (i.e., when transient data is missing), this could actually be non-local as well.